AK/BitCast.h

```diff
@@ -34,7 +40,7 @@ inline T bit_cast(const U& a)
     static_assert(sizeof(T) == sizeof(U));
 
     T result;
-    __builtin_memcpy(&result, &a, sizeof(T));
+    memcpy((void*)&result, &a, sizeof(T));
```

Comment:
Why is this necessary? Does it change the generated code at all?

Reply:
haven't checked that, but I just happened  to stumble on that while trying to use bit_cast's for long double conversion and I thought the "zen" was to try write every method on our own, and not rely on compilers and their libraries if possible

the (void*) cast is useless though

Reply:
And looking at [this](https://godbolt.org/z/9jqd5GfWc) it seems that buildins are better due to being optimized better to just to movs instead of a rep movsb and allways inlined, maybe we can apply some of these optimizations too in out libC implementation

Reply:
guess I'll drop that change for now